packages/order-utils/src/rate_utils.ts

```diff
@@ -0,0 +1,48 @@
+import { schemas } from '@0xproject/json-schemas';
+import { SignedOrder } from '@0xproject/types';
+import { BigNumber } from '@0xproject/utils';
+
+import { assert } from './assert';
+import { constants } from './constants';
+
+export const rateUtils = {
+    /**
+     * Takes a signed order and calculates the fee adjusted rate (takerAsset/makerAsset) by calculating how much takerAsset
+     * is required to cover the fees (feeRate * takerFee), adding the takerAssetAmount and dividing by makerAssetAmount
+     * @param   signedOrder     An object that conforms to the signedOrder interface
+     * @param   feeRate         The market rate of ZRX denominated in takerAssetAmount
+     *                          (ex. feeRate is 0.1 takerAsset/ZRX if it takes 1 unit of takerAsset to buy 10 ZRX)
+     *                          Defaults to 0
+     * @return  The rate (takerAsset/makerAsset) of the order adjusted for fees
+     */
+    getFeeAdjustedRateOfOrder(signedOrder: SignedOrder, feeRate: BigNumber = constants.ZERO_AMOUNT): BigNumber {
```

Comment:
hmm yea good call, I had some issues with some other functions that took in orders and had to return orders, in those cases you actually have to use generics so that the type system knows a SignedOrder should be spit out if a SignedOrder is passed in, let me see if I can finagle it